src/data/ability.ts

```diff
@@ -3594,7 +3594,15 @@ export class PostTurnStatStageChangeAbAttr extends PostTurnAbAttr {
 
   applyPostTurn(pokemon: Pokemon, passive: boolean, simulated: boolean, args: any[]): boolean {
     if (!simulated) {
-      pokemon.scene.unshiftPhase(new StatStageChangePhase(pokemon.scene, pokemon.getBattlerIndex(), true, this.stats, this.stages));
+      if (!pokemon.hasAbility(Abilities.SPEED_BOOST)) {
+        pokemon.scene.unshiftPhase(new StatStageChangePhase(pokemon.scene, pokemon.getBattlerIndex(), true, this.stats, this.stages));
+        this.showAbility = true;
+      } else if (pokemon.hasAbility(Abilities.SPEED_BOOST) && !pokemon.turnData.switchedInThisTurn && !pokemon.turnData.failedRunAway) {
+        pokemon.scene.unshiftPhase(new StatStageChangePhase(pokemon.scene, pokemon.getBattlerIndex(), true, this.stats, this.stages));
+        this.showAbility = true;
+      } else {
+        this.showAbility = false;
+      }
```

Comment:
```suggestion
      if (!pokemon.hasAbility(Abilities.SPEED_BOOST) || (!pokemon.turnData.switchedInThisTurn && !pokemon.turnData.failedRunAway)) {
        if (!simulated) {
           pokemon.scene.unshiftPhase(new StatStageChangePhase(pokemon.scene, pokemon.getBattlerIndex(), true, this.stats, this.stages));
        }
        return true;
      } 
      return false;
```
You should remove the `if (!simulated)` block encapsulating this, too... GitHub is being annoying about me removing it lol